### PR TITLE
CP-52458: Remove BIOS related code from host-installer

### PR DESCRIPTION
**Pull request by stephenchengCloud**
XS9 only supports UEFI boot mode, so remove BIOS related code from host-installer.
Test:
- Manual installation:
image
image
- XenRT tests:
4154212 UpgradePluginHTTP
4154213 ring0upg
207934 Ring0 BST

### CODE REVIEW

**Comment by freddy77** [comment]
Please add some explanation to the commit message, this is supposed to be an open source project, the reference to an internal ticket system accounts to nothing.

One thing to state is that also XS8 requires UEFI so upgrades and restores will continue to work.

File: backend.py [hunk 1]
```diff
@@ -534,29 +535,23 @@ def partitionTargetDisk(disk, existing, preserve_first_partition, create_sr_part
...
-        # Determine target install's boot mode and boot partition number
-        target_boot_mode = TARGET_BOOT_MODE_LEGACY
+        # Determine target install's boot partition number
         if existing.boot_device:
             boot_partnum = tool.partitionNumber(existing.boot_device)
             boot_part = tool.getPartition(boot_partnum)
-            if 'id' in boot_part and boot_part['id'] == GPTPartitionTool.ID_EFI_BOOT:
-                target_boot_mode = TARGET_BOOT_MODE_UEFI
+            if 'id' not in boot_part or boot_part['id'] != GPTPartitionTool.ID_EFI_BOOT:
+                raise RuntimeError("Boot partition is not set up for UEFI mode or missing EFI partition ID.")
```
[the remark this answers is not in the document]
Comment (stephenchengCloud) [reply backend.py:543]: OK. So we don't support upgrading from a BIOS installation to a UEFI installation.

[the remark this answers is not in the document]
Comment (stephenchengCloud) [reply backend.py:543]: I found that `partitionTargetDisk` is already a step of upgrading, so this check is already in upgrading.
If we want to advance this check to `isUpgradeable()`, we have to call `partitionTargetDisk` repeatedly in `isUpgradeable()`.

**Comment by stephenchengCloud** [comment]
> Please add some explanation to the commit message, this is supposed to be an open source project, the reference to an internal ticket system accounts to nothing.

You're right. My fault. Will add come explanation.

**Comment by stephenchengCloud** [comment]
Tested:
4157523  ring0upg
4157526  UpgradePluginHTTP

File: backend.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rosslagerwall) [reply backend.py:538]: Yes, they are both called in the upgrade path. My point was that adding a check to `isUpgradeable()` prevents offering to the user to upgrade an installation that we know can't be upgraded and then later failing during the upgrade with the above error.

I don't think you need to repeat all of partitionTargetDisk(). You've already got access to the boot device in `isUpgradeable()` so you just need to check its id / type code.

**Comment by stephenchengCloud** [comment]
> This looks fine to me except for the outstanding point about `isUpgradeable()` and removal of `bootloader/isolinux.cfg`.

Thanks for your comments. All comments are resolved.

**Comment by stephenchengCloud** [comment]
Tested the `upgrade` scenario after the code updated:
image
image

**Comment by rosslagerwall** [comment]
> > This looks fine to me except for the outstanding point about `isUpgradeable()` and removal of `bootloader/isolinux.cfg`.
> 
> Thanks for your comments. All comments are resolved.

You still need to removal the actual file, not just the reference in the `Makefile`. i.e.

```
git rm bootloader/isolinux.cfg
```

**Review by liulinC — APPROVED** [review]

**Pull request merged**